### PR TITLE
More Omnipod updates & cleanup

### PR DESCRIPTION
**Pull request by bartsopers**
Also includes 2 bug fixes in ValidatingEditTextPreference that I originally submitted in PR #2947 which I now closed in favor of this PR.

### CODE REVIEW

File: omnipod/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dagger/OmnipodModule.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    // Service
+    @ContributesAndroidInjector
+    abstract fun omnipodCommunicationManagerProvider(): OmnipodRileyLinkCommunicationManager
+    @ContributesAndroidInjector abstract fun aapsOmnipodManagerProvider(): AapsOmnipodManager
+
+    // Data
+    @ContributesAndroidInjector abstract fun initPodRefreshAction(): InitPodRefreshAction
+    @ContributesAndroidInjector abstract fun podStateManager(): PodStateManager
+    @ContributesAndroidInjector abstract fun initPodTask(): InitPodTask
+    @ContributesAndroidInjector abstract fun omnipodPumpPlugin(): OmnipodPumpPlugin
```
Comment (AdrianLxM) [inline omnipod/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dagger/OmnipodModule.kt:43]: This is a workaround until the plugin is ready?

**Review by AdrianLxM — CHANGES_REQUESTED** [review]
Nice large PR :)

I have a few comments in the general part of this PR.
I only reviewed the parts that intertwine with the rest of the app. The core omnipod packages I did not have a deeper look.

**Comment by AdrianLxM** [comment]
As I don't want to force my opinion on how `lateinit` (avoid unless really necessary ;) ): Here are my ideas as PR back to your branch: https://github.com/AAPS-Omnipod/AndroidAPS/pull/166

Feel free to use that - or discard it.

File: omnipod/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dagger/OmnipodModule.kt (hunk 1 above)
Comment (bartsopers) [reply omnipod/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dagger/OmnipodModule.kt:43]: What do you mean?

File: app/src/main/java/info/nightscout/androidaps/queue/commands/CommandStartPump.kt [hunk 2]
```diff
@@ -12,12 +14,19 @@ class CommandStartPump(
...
 
     override fun execute() {
         val pump = activePlugin.activePump
         if (pump is LocalInsightPlugin) {
             val result = pump.startPump()
             callback?.result(result)?.run()
+        } else if (pump.pumpDescription.pumpType == PumpType.Insulet_Omnipod) {
+            // When using CommandQueue.setProfile, it refuses to set the profile is the same as the current profile
+            // However we need to set the current profile to resume delivery in case the Pod is suspended
+            if (profileFunction.getProfile() != null) {
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply app/src/main/java/info/nightscout/androidaps/queue/commands/CommandStartPump.kt]: I decided to just pass the profile, even if it's null, as AapsOmnipodManager will handle a null profile and show a notification.

File: core/src/main/java/info/nightscout/androidaps/plugins/general/overview/notifications/Notification.java [hunk 3]
```diff
@@ -72,6 +72,8 @@
     public static final int OMNIPOD_POD_NOT_ATTACHED = 59;
     public static final int CARBS_REQUIRED = 60;
     public static final int IMPORTANCE_HIGH = 2;
+    public static final int OMNIPOD_POD_SUSPENDED = 61;
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply core/src/main/java/info/nightscout/androidaps/plugins/general/overview/notifications/Notification.java:74]: Done!

File: core/src/main/java/info/nightscout/androidaps/utils/textValidator/ValidatingEditTextPreference.kt [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
 import android.util.AttributeSet
 import androidx.preference.EditTextPreference
-import androidx.preference.EditTextPreference.OnBindEditTextListener
 import androidx.preference.PreferenceViewHolder
 import info.nightscout.androidaps.core.R
 
 class ValidatingEditTextPreference(ctx: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int)
     : EditTextPreference(ctx, attrs, defStyleAttr, defStyleRes) {
 
+    private lateinit var validatorParameters: DefaultEditTextValidator.Parameters
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply core/src/main/java/info/nightscout/androidaps/utils/textValidator/ValidatingEditTextPreference.kt]: Merged your PR so this has been fixed

File: omnipod/src/test/java/info/nightscout/androidaps/plugins/pump/omnipod/manager/AapsPodStateManagerTest.java [hunk 5]
```diff
@@ -24,8 +22,7 @@
 public class AapsPodStateManagerTest {
     @Mock AAPSLogger aapsLogger;
     @Mock SP sp;
-    @Mock OmnipodPumpStatus omnipodPumpStatus;
-    RxBusWrapper rxBus = new RxBusWrapper();
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply omnipod/src/test/java/info/nightscout/androidaps/plugins/pump/omnipod/manager/AapsPodStateManagerTest.java:28]: I changed this to just instantiating RxBusWrapper

File: core/src/main/java/info/nightscout/androidaps/utils/textValidator/DefaultEditTextValidator.kt [hunk 6]
```diff
@@ -248,4 +242,19 @@ class DefaultEditTextValidator : EditTextValidator {
         } catch (e: Throwable) {
             !TextUtils.isEmpty(editTextView.error)
         }
+
+    class Parameters {
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply core/src/main/java/info/nightscout/androidaps/utils/textValidator/DefaultEditTextValidator.kt]: Merged your PR so this has been fixed

**Review by AdrianLxM — APPROVED** [review]
LGTM!

**Pull request merged**